E0618 generates invalid suggestion / zero length span

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code

```Rust
enum Enum<T> { , SVariant { v: T }, UVariant }

macro_rules! is_variant {
    (TSVariant, ) => (!);
    (SVariant, ) => (!);
    (UVariant, $expr:expr) => (is_variant!(@check UVariant, {}, $expr));
    (@check $variant:ident, $matcher:tt, $expr:expr) => (
        assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
                );
    );
}

fn main() {
    is_variant!(UVariant, Enum::<()>::UVariant);
}

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
307c573d57616a9f7500168061ce6fdbb5060a58
```

### Error output

```
....
error[E0618]: expected function, found enum variant `Enum<()>::UVariant`
  --> ./658FADDF1913874E759F16E9741008D4DCD6F922030C6069C1DFCB1FD8483AEB.rs:14:27
   |
1  | enum Enum<T> { , SVariant { v: T }, UVariant }
   |                                     -------- enum variant `Enum<()>::UVariant` defined here
...
8  |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
   |                                                        -------- call expression requires function
...
14 |     is_variant!(UVariant, Enum::<()>::UVariant);
   |                           ^^^^^^^^^^^^^^^^^^^^
   |
help: `Enum<()>::UVariant` is a unit enum variant, and does not take parentheses to be constructed
   |
8  |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
   |

error: aborting due to 3 previous errors
```

For line 8 there is a zero-size span somewhere pointing at nothing essentially 

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:706:9:
Span must not be empty and have no suggestion
stack backtrace:
   0:     0x7f8b056f3f71 - std::backtrace_rs::backtrace::libunwind::trace::h710b066054616472
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f8b056f3f71 - std::backtrace_rs::backtrace::trace_unsynchronized::had82bb625e151070
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f8b056f3f71 - std::sys_common::backtrace::_print_fmt::h053a03662b469137
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f8b056f3f71 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h32f52161b81de05f
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f8b0573cfa7 - core::fmt::rt::Argument::fmt::h1dc8855cf0be02de
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/rt.rs:138:9
   5:     0x7f8b0573cfa7 - core::fmt::write::h35c6f281391d7802
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f8b056e7a05 - std::io::Write::write_fmt::hd9ec8ebef5dd5588
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/io/mod.rs:1714:15
   7:     0x7f8b056f3d44 - std::sys_common::backtrace::_print::h5a362b959b8e7b59
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f8b056f3d44 - std::sys_common::backtrace::print::h375551929bb59c0d
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f8b056a818e - std::panicking::panic_hook_with_disk_dump::{{closure}}::hfbd0b2a743fee161
  10:     0x7f8b056a7ebd - std::panicking::panic_hook_with_disk_dump::ha7fe0c75495a15b9
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:312:9
  11:     0x7f8b0726d580 - rustc_driver_impl[4105f87af516559e]::install_ice_hook::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:1384:13
  12:     0x7f8b056a8a37 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h1e0ac7fb46b75191
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2021:9
  13:     0x7f8b056a8a37 - std::panicking::rust_panic_with_hook::h389e8fc3aeb91007
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:733:13
  14:     0x7f8b056f43d1 - std::panicking::begin_panic_handler::{{closure}}::hbf4f1fefb5b0fa52
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:619:13
  15:     0x7f8b056f4206 - std::sys_common::backtrace::__rust_end_short_backtrace::h49ec237e0a1755fc
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:170:18
  16:     0x7f8b056a85a2 - rust_begin_unwind
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:617:5
  17:     0x7f8b0567b763 - core::panicking::panic_fmt::h2aa9ff874ea77c22
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:67:14
  18:     0x7f8b07a4d464 - <rustc_errors[d21bc24ba71585f]::diagnostic::Diagnostic>::span_suggestion_with_style::<alloc[89f1688d944dadaf]::string::String, &str>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:706:9
  19:     0x7f8b077eae0e - <rustc_errors[d21bc24ba71585f]::diagnostic::Diagnostic>::span_suggestion_verbose::<alloc[89f1688d944dadaf]::string::String, &str>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:729:9
  20:     0x7f8b077eae0e - <rustc_errors[d21bc24ba71585f]::diagnostic_builder::DiagnosticBuilder<rustc_span[97ecb726a3b0b220]::ErrorGuaranteed>>::span_suggestion_verbose::<alloc[89f1688d944dadaf]::string::String, &str>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic_builder.rs:466:35
  21:     0x7f8b077eae0e - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::report_invalid_callee
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/callee.rs:632:13
  22:     0x7f8b077eae0e - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::confirm_builtin_call
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/callee.rs:439:27
  23:     0x7f8b077e710e - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_call
  24:     0x7f8b07884613 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:323:45
  25:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  26:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  27:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  28:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  29:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  30:     0x7f8b078313e1 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:128:9
  31:     0x7f8b078313e1 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:118:18
  32:     0x7f8b078313e1 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_decl_initializer
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1444:13
  33:     0x7f8b07831642 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_decl
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1455:27
  34:     0x7f8b078842c6 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_let
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:1204:9
  35:     0x7f8b078842c6 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:314:40
  36:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  37:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  38:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  39:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  40:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  41:     0x7f8b07884415 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_has_type_or_error::<<rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_then_else::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:69:22
  42:     0x7f8b07884415 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_then_else
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:1012:23
  43:     0x7f8b07884415 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:335:17
  44:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  45:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  46:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  47:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  48:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  49:     0x7f8b07885f5b - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_unary
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:368:27
  50:     0x7f8b07885f5b - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:285:45
  51:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  52:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  53:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  54:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  55:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  56:     0x7f8b07884fcf - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:337:39
  57:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  58:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  59:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  60:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  61:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  62:     0x7f8b07884415 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_has_type_or_error::<<rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_then_else::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:69:22
  63:     0x7f8b07884415 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_then_else
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:1012:23
  64:     0x7f8b07884415 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:335:17
  65:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  66:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  67:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  68:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  69:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  70:     0x7f8b07831bc4 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:149:9
  71:     0x7f8b07831bc4 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_stmt
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1518:17
  72:     0x7f8b0783219d - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1570:17
  73:     0x7f8b0783219d - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::with_breakable_ctxt::<<rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:1535:22
  74:     0x7f8b0783219d - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_block_with_expected
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1568:26
  75:     0x7f8b078841fe - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:322:41
  76:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:226:18
  77:     0x7f8b078044b2 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  78:     0x7f8b078044b2 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::ty::Ty, <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  79:     0x7f8b078044b2 - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:222:18
  80:     0x7f8b0788387a - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:176:9
  81:     0x7f8b07805fbf - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:128:9
  82:     0x7f8b07805fbf - <rustc_hir_typeck[758104400a144285]::fn_ctxt::FnCtxt>::check_return_expr
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:815:30
  83:     0x7f8b07a7100f - rustc_hir_typeck[758104400a144285]::check::check_fn
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/check.rs:117:5
  84:     0x7f8b07946e62 - rustc_hir_typeck[758104400a144285]::typeck_with_fallback::<rustc_hir_typeck[758104400a144285]::typeck::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:202:9
  85:     0x7f8b07946e62 - rustc_hir_typeck[758104400a144285]::typeck_with_fallback::<rustc_hir_typeck[758104400a144285]::typeck::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:161:1
  86:     0x7f8b07946e62 - rustc_hir_typeck[758104400a144285]::typeck
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:148:5
  87:     0x7f8b08d47883 - rustc_query_impl[e8f6d07168448de3]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:582:47
  88:     0x7f8b08d47883 - rustc_query_impl[e8f6d07168448de3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[e8f6d07168448de3]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:506:18
  89:     0x7f8b09032ecb - rustc_query_impl[e8f6d07168448de3]::query_impl::typeck::dynamic_query::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:578:25
  90:     0x7f8b09032ecb - <rustc_query_impl[e8f6d07168448de3]::query_impl::typeck::dynamic_query::{closure#2} as core[1cc32ee8b476061a]::ops::function::FnOnce<(rustc_middle[2732fee5f659919]::ty::context::TyCtxt, rustc_span[97ecb726a3b0b220]::def_id::LocalDefId)>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/ops/function.rs:250:5
  91:     0x7f8b08ce7a1d - <rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false> as rustc_query_system[9e813e950960ba8f]::query::config::QueryConfig<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:117:9
  92:     0x7f8b08ce7a1d - rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:72
  93:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::enter_context::<rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:82:9
  94:     0x7f8b08ce7a1d - <std[4f10ea13563ec26d]::thread::local::LocalKey<core[1cc32ee8b476061a]::cell::Cell<*const ()>>>::try_with::<rustc_middle[2732fee5f659919]::ty::context::tls::enter_context<rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:270:16
  95:     0x7f8b08ce7a1d - <std[4f10ea13563ec26d]::thread::local::LocalKey<core[1cc32ee8b476061a]::cell::Cell<*const ()>>>::with::<rustc_middle[2732fee5f659919]::ty::context::tls::enter_context<rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:246:9
  96:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::enter_context::<rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:79:9
  97:     0x7f8b08ce7a1d - <rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query::<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:149:13
  98:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::with_related_context::<<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:133:9
  99:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::with_context::<rustc_middle[2732fee5f659919]::ty::context::tls::with_related_context<<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:36
 100:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::with_context_opt::<rustc_middle[2732fee5f659919]::ty::context::tls::with_context<rustc_middle[2732fee5f659919]::ty::context::tls::with_related_context<<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:100:18
 101:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::with_context::<rustc_middle[2732fee5f659919]::ty::context::tls::with_related_context<<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:5
 102:     0x7f8b08ce7a1d - rustc_middle[2732fee5f659919]::ty::context::tls::with_related_context::<<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:124:5
 103:     0x7f8b08ce7a1d - <rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt as rustc_query_system[9e813e950960ba8f]::query::QueryContext>::start_query::<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:134:9
 104:     0x7f8b08ce7a1d - rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:18
 105:     0x7f8b08ce7a1d - rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:414:9
 106:     0x7f8b08ce7a1d - rustc_query_system[9e813e950960ba8f]::query::plumbing::try_execute_query::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:359:13
 107:     0x7f8b08e918b6 - rustc_query_system[9e813e950960ba8f]::query::plumbing::get_query_non_incr::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:811:32
 108:     0x7f8b08e918b6 - stacker[13632619fe1702]::maybe_grow::<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::get_query_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
 109:     0x7f8b08e918b6 - rustc_data_structures[3dc85bbe0d20d026]::stack::ensure_sufficient_stack::<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[9e813e950960ba8f]::query::plumbing::get_query_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
 110:     0x7f8b08e918b6 - rustc_query_system[9e813e950960ba8f]::query::plumbing::get_query_non_incr::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:811:5
 111:     0x7f8b08e918b6 - rustc_query_impl[e8f6d07168448de3]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:556:26
 112:     0x7f8b07665f5d - rustc_middle[2732fee5f659919]::query::plumbing::query_ensure::<rustc_query_system[9e813e950960ba8f]::query::caches::VecCache<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 8usize]>>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:171:9
 113:     0x7f8b07665f5d - <rustc_middle[2732fee5f659919]::query::plumbing::TyCtxtEnsure>::typeck::<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:346:17
 114:     0x7f8b07665f5d - rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/lib.rs:247:13
 115:     0x7f8b07665f5d - <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners::<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/hir/map/mod.rs:492:82
 116:     0x7f8b07665f5d - rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in::<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:218:66
 117:     0x7f8b07665f5d - <core[1cc32ee8b476061a]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}> as core[1cc32ee8b476061a]::ops::function::FnOnce<()>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panic/unwind_safe.rs:271:9
 118:     0x7f8b076420ab - std[4f10ea13563ec26d]::panicking::try::do_call::<core[1cc32ee8b476061a]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:524:40
 119:     0x7f8b076420ab - std[4f10ea13563ec26d]::panicking::try::<(), core[1cc32ee8b476061a]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:488:19
 120:     0x7f8b076420ab - std[4f10ea13563ec26d]::panic::catch_unwind::<core[1cc32ee8b476061a]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panic.rs:142:14
 121:     0x7f8b076420ab - rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in::<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:218:33
 122:     0x7f8b076420ab - <core[1cc32ee8b476061a]::slice::iter::Iter<rustc_span[97ecb726a3b0b220]::def_id::LocalDefId> as core[1cc32ee8b476061a]::iter::traits::iterator::Iterator>::for_each::<rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/slice/iter/macros.rs:254:21
 123:     0x7f8b076420ab - rustc_data_structures[3dc85bbe0d20d026]::sync::par_for_each_in::<&[rustc_span[97ecb726a3b0b220]::def_id::LocalDefId], <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:217:13
 124:     0x7f8b07633148 - <rustc_middle[2732fee5f659919]::hir::map::Map>::par_body_owners::<rustc_hir_analysis[2ac1febfb8d48090]::check_crate::{closure#7}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/hir/map/mod.rs:492:9
 125:     0x7f8b07633148 - rustc_hir_analysis[2ac1febfb8d48090]::check_crate
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/lib.rs:244:5
 126:     0x7f8b07dbccf4 - rustc_interface[15bc44e426197ea8]::passes::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/passes.rs:761:5
 127:     0x7f8b08d47e31 - rustc_query_impl[e8f6d07168448de3]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:582:47
 128:     0x7f8b08d47e31 - rustc_query_impl[e8f6d07168448de3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[e8f6d07168448de3]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 1usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:506:18
 129:     0x7f8b08f9d4f7 - rustc_query_impl[e8f6d07168448de3]::query_impl::analysis::dynamic_query::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:578:25
 130:     0x7f8b08f9d4f7 - <rustc_query_impl[e8f6d07168448de3]::query_impl::analysis::dynamic_query::{closure#2} as core[1cc32ee8b476061a]::ops::function::FnOnce<(rustc_middle[2732fee5f659919]::ty::context::TyCtxt, ())>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/ops/function.rs:250:5
 131:     0x7f8b08c5b5d0 - <rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::SingleCache<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 1usize]>>, false, false, false> as rustc_query_system[9e813e950960ba8f]::query::config::QueryConfig<rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:117:9
 132:     0x7f8b08c5b5d0 - rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr::<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::SingleCache<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:72
 133:     0x7f8b08c5b5d0 - rustc_middle[2732fee5f659919]::ty::context::tls::enter_context::<rustc_query_system[9e813e950960ba8f]::query::plumbing::execute_job_non_incr<rustc_query_impl[e8f6d07168448de3]::DynamicConfig<rustc_query_system[9e813e950960ba8f]::query::caches::SingleCache<rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[e8f6d07168448de3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[2732fee5f659919]::query::erase::Erased<[u8; 1usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:82:9
 .....

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please attach the file at `/home/matthias/vcs/github/rust_codegen_grouped/src/test/ui/icemaker_reduced/SPAN/interesting/rustc-ice-2023-08-10T15:31:10.656994812Z-1958175.txt` to your bug report

query stack during panic:
#0 [typeck] type-checking `main`
#1 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 2 previous errors
```

</p>
</details>

--- a/compiler/rustc_hir_typeck/src/callee.rs
+++ b/compiler/rustc_hir_typeck/src/callee.rs
@@ -599,6 +599,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 = self.typeck_results.borrow().qpath_res(qpath, callee_expr.hir_id)
             // Only suggest removing parens if there are no arguments
             && arg_exprs.is_empty()
+            && call_expr.span.contains(callee_expr.span)
         {
             let descr = match kind {
                 def::CtorOf::Struct => "struct",

--- a/tests/ui/suggestions/issue-114701.rs
+++ b/tests/ui/suggestions/issue-114701.rs
@@ -1,0 +1,15 @@
+enum Enum<T> { , SVariant { v: T }, UVariant } //~ ERROR expected identifier, found `,`
+
+macro_rules! is_variant {
+    (TSVariant, ) => (!);
+    (SVariant, ) => (!);
+    (UVariant, $expr:expr) => (is_variant!(@check UVariant, {}, $expr));
+    (@check $variant:ident, $matcher:tt, $expr:expr) => (
+        assert!(if let Enum::$variant::<()> $matcher = $expr () else { false }, //~ ERROR this `if` expression
+                );
+    );
+}
+
+fn main() {
+    is_variant!(UVariant, Enum::<()>::UVariant); //~ ERROR expected function
+}

--- a/tests/ui/suggestions/issue-114701.rs
+++ b/tests/ui/suggestions/issue-114701.rs
@@ -1,11 +1,11 @@
-enum Enum<T> { , SVariant { v: T }, UVariant } //~ ERROR expected identifier, found `,`
+enum Enum<T> { SVariant { v: T }, UVariant }
 
 macro_rules! is_variant {
     (TSVariant, ) => (!);
     (SVariant, ) => (!);
     (UVariant, $expr:expr) => (is_variant!(@check UVariant, {}, $expr));
     (@check $variant:ident, $matcher:tt, $expr:expr) => (
-        assert!(if let Enum::$variant::<()> $matcher = $expr () else { false }, //~ ERROR this `if` expression
+        assert!(if let Enum::$variant::<()> $matcher = $expr () { true } else { false },
                 );
     );
 }

--- a/tests/ui/suggestions/issue-114701.stderr
+++ b/tests/ui/suggestions/issue-114701.stderr
@@ -1,48 +1,15 @@
-error: expected identifier, found `,`
-  --> $DIR/issue-114701.rs:1:16
-   |
-LL | enum Enum<T> { , SVariant { v: T }, UVariant }
-   |                ^
-   |                |
-   |                expected identifier
-   |                help: remove this comma
-
-error: this `if` expression is missing a block after the condition
-  --> $DIR/issue-114701.rs:8:17
-   |
-LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
-   |                 ^^
-...
-LL |     is_variant!(UVariant, Enum::<()>::UVariant);
-   |     ------------------------------------------- in this macro invocation
-   |
-help: add a block here
-  --> $DIR/issue-114701.rs:8:64
-   |
-LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
-   |                                                                ^
-...
-LL |     is_variant!(UVariant, Enum::<()>::UVariant);
-   |     ------------------------------------------- in this macro invocation
-   = note: this error originates in the macro `is_variant` (in Nightly builds, run with -Z macro-backtrace for more info)
-help: remove the `if` if you meant to write a `let...else` statement
-   |
-LL -         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
-LL +         assert!(let Enum::$variant::<()> $matcher = $expr () else { false },
-   |
-
 error[E0618]: expected function, found `Enum<()>`
   --> $DIR/issue-114701.rs:14:27
    |
-LL | enum Enum<T> { , SVariant { v: T }, UVariant }
-   |                                     -------- `Enum::UVariant` defined here
+LL | enum Enum<T> { SVariant { v: T }, UVariant }
+   |                                   -------- `Enum::UVariant` defined here
 ...
-LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
+LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () { true } else { false },
    |                                                        -------- call expression requires function
 ...
 LL |     is_variant!(UVariant, Enum::<()>::UVariant);
    |                           ^^^^^^^^^^^^^^^^^^^^
 
-error: aborting due to 3 previous errors
+error: aborting due to previous error
 
 For more information about this error, try `rustc --explain E0618`.

--- a/tests/ui/suggestions/issue-114701.stderr
+++ b/tests/ui/suggestions/issue-114701.stderr
@@ -1,0 +1,48 @@
+error: expected identifier, found `,`
+  --> $DIR/issue-114701.rs:1:16
+   |
+LL | enum Enum<T> { , SVariant { v: T }, UVariant }
+   |                ^
+   |                |
+   |                expected identifier
+   |                help: remove this comma
+
+error: this `if` expression is missing a block after the condition
+  --> $DIR/issue-114701.rs:8:17
+   |
+LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
+   |                 ^^
+...
+LL |     is_variant!(UVariant, Enum::<()>::UVariant);
+   |     ------------------------------------------- in this macro invocation
+   |
+help: add a block here
+  --> $DIR/issue-114701.rs:8:64
+   |
+LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
+   |                                                                ^
+...
+LL |     is_variant!(UVariant, Enum::<()>::UVariant);
+   |     ------------------------------------------- in this macro invocation
+   = note: this error originates in the macro `is_variant` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: remove the `if` if you meant to write a `let...else` statement
+   |
+LL -         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
+LL +         assert!(let Enum::$variant::<()> $matcher = $expr () else { false },
+   |
+
+error[E0618]: expected function, found `Enum<()>`
+  --> $DIR/issue-114701.rs:14:27
+   |
+LL | enum Enum<T> { , SVariant { v: T }, UVariant }
+   |                                     -------- `Enum::UVariant` defined here
+...
+LL |         assert!(if let Enum::$variant::<()> $matcher = $expr () else { false },
+   |                                                        -------- call expression requires function
+...
+LL |     is_variant!(UVariant, Enum::<()>::UVariant);
+   |                           ^^^^^^^^^^^^^^^^^^^^
+
+error: aborting due to 3 previous errors
+
+For more information about this error, try `rustc --explain E0618`.

